core/station/impl/src/mappers/request_operation.rs

```diff
@@ -444,6 +445,58 @@ impl From<SystemUpgradeOperation> for station_api::SystemUpgradeOperationDTO {
     }
 }
 
+impl From<SystemRestoreTarget> for station_api::SystemRestoreTargetDTO {
+    fn from(value: SystemRestoreTarget) -> Self {
+        match value {
+            SystemRestoreTarget::RestoreStation => {
+                station_api::SystemRestoreTargetDTO::RestoreStation
+            }
+            SystemRestoreTarget::RestoreUpgrader => {
+                station_api::SystemRestoreTargetDTO::RestoreUpgrader
+            }
+        }
+    }
+}
+
+impl From<station_api::SystemRestoreTargetDTO> for SystemRestoreTarget {
+    fn from(value: station_api::SystemRestoreTargetDTO) -> Self {
+        match value {
+            station_api::SystemRestoreTargetDTO::RestoreStation => {
+                SystemRestoreTarget::RestoreStation
+            }
+            station_api::SystemRestoreTargetDTO::RestoreUpgrader => {
+                SystemRestoreTarget::RestoreUpgrader
+            }
+        }
+    }
+}
+
+impl From<SystemRestoreOperationInput> for station_api::SystemRestoreOperationInput {
+    fn from(input: SystemRestoreOperationInput) -> station_api::SystemRestoreOperationInput {
+        station_api::SystemRestoreOperationInput {
+            target: input.target.into(),
+            snapshot_id: hex::encode(&input.snapshot_id),
+        }
+    }
+}
+
+impl From<station_api::SystemRestoreOperationInput> for SystemRestoreOperationInput {
+    fn from(input: station_api::SystemRestoreOperationInput) -> SystemRestoreOperationInput {
+        SystemRestoreOperationInput {
+            target: input.target.into(),
+            snapshot_id: hex::decode(&input.snapshot_id).unwrap(),
```

Comment:
please use `expect` or `unwrap_or_else`->`trap` so that you can provide an error message